Archipelago.MultiClient.Net/Helpers/ArchipelagoSocketHelper.cs

```diff
@@ -229,7 +238,75 @@ public void SendMultiplePacketsAsync(Action<bool> onComplete = null, params Arch
                 throw new ArchipelagoSocketClosedException();
             }
         }
+#else
+        /// <summary>
+        ///     Send a single <see cref="ArchipelagoPacketBase"/> derived packet asynchronously.
+        /// </summary>
+        /// <param name="packet">
+        ///     The packet to send to the server.
+        /// </param>
+        /// <exception cref="T:Archipelago.MultiClient.Net.Exceptions.ArchipelagoSocketClosedException">
+        ///     The websocket connection is not alive.
+        /// </exception>
+        public async Task<bool> SendPacketAsync(ArchipelagoPacketBase packet)
+        {
+            return await SendMultiplePacketsAsync(new List<ArchipelagoPacketBase> { packet });
+        }
+
+        /// <summary>
+        ///     Send a single <see cref="ArchipelagoPacketBase"/> derived packet asynchronously.
+        /// </summary>
+        /// <param name="packets">
+        ///     The packets to send to the server.
+        /// </param>
+        /// <remarks>
+        ///     The packets will be sent in the order they are provided in the list.
+        /// </remarks>
+        /// <exception cref="T:Archipelago.MultiClient.Net.Exceptions.ArchipelagoSocketClosedException">
+        ///     The websocket connection is not alive.
+        /// </exception>
+        public async Task<bool> SendMultiplePacketsAsync(List<ArchipelagoPacketBase> packets)
+        {
+            return await SendMultiplePacketsAsync(packets.ToArray());
+        }
+
+        /// <summary>
+        ///     Send a single <see cref="ArchipelagoPacketBase"/> derived packet asynchronously.
+        /// </summary>
+        /// <param name="packets">
+        ///     The packets to send to the server.
+        /// </param>
+        /// <remarks>
+        ///     The packets will be sent in the order they are provided as arguments.
+        /// </remarks>
+        /// <exception cref="T:Archipelago.MultiClient.Net.Exceptions.ArchipelagoSocketClosedException">
+        ///     The websocket connection is not alive.
+        /// </exception>
+        public async Task<bool> SendMultiplePacketsAsync(params ArchipelagoPacketBase[] packets)
+        {
+            var taskCompletionSource = new TaskCompletionSource<bool>();
 
+            if (webSocket.IsAlive)
+            {
+                var packetAsJson = JsonConvert.SerializeObject(packets);
+                webSocket.SendAsync(packetAsJson, success => {
+                    taskCompletionSource.SetResult(success);
+                });
+
+                if (PacketsSent != null)
+                {
+                    PacketsSent(packets);
+                }
+            }
+            else
+            {
+                taskCompletionSource.SetException(new ArchipelagoSocketClosedException());
+            }
+
+            return await taskCompletionSource.Task;
```

Comment:
```suggestion
        public Task<bool> SendMultiplePacketsAsync(params ArchipelagoPacketBase[] packets)
        {
            var taskCompletionSource = new TaskCompletionSource<bool>();

            if (webSocket.IsAlive)
            {
                var packetAsJson = JsonConvert.SerializeObject(packets);
                webSocket.SendAsync(packetAsJson, success => {
                    taskCompletionSource.SetResult(success);
                });

                if (PacketsSent != null)
                {
                    PacketsSent(packets);
                }
            }
            else
            {
                taskCompletionSource.SetException(new ArchipelagoSocketClosedException());
            }

            return taskCompletionSource.Task;
```

What do you think about this: so that the caller can await the task completion of the packet sending instead of it being awaited in this function?